src/server/run_mir.cpp

```diff
@@ -110,7 +221,20 @@ void mir::run_mir(
             {
                 for (auto sig : intercepted)
                 {
-                    old_handler[sig] = signal(sig, fatal_signal_cleanup);
+                    struct sigaction sig_handler_desc;
+                    sig_handler_desc.sa_flags = SA_SIGINFO;
+                    sig_handler_desc.sa_sigaction = &fatal_signal_cleanup;
+
+                    if (sigaction(sig, &sig_handler_desc, old_handlers.initialiser_for_signal(sig)))
+                    {
+                        using namespace std::string_literals;
+                        BOOST_THROW_EXCEPTION((
+                            std::system_error{
+                                errno,
+                                std::system_category(),
+                                "Failed to install signal handler for "s + signum_to_string(sig)
+                            }));
```

Comment:
Is throwing an exception actually useful?

Reply:
As opposed to simply `fatal_error_abort()`? Unlike the usages above, here we're in a context where there's going to be a catch handler further up the stack, and throwing gets us a nicer error and lets consuming code do something before dying.

If the alternative is not checking the return of `sigaction()` then that's maybe ok; if I understand it correctly that call can only fail in cases where we've made a programmer error. On the other hand, I *like* getting told when I've made a programming error ;)